File: src/main/java/io/github/flegacy/flms/mining/WorldProtectionListener.java
package io.github.flegacy.flms.mining;

import org.bukkit.GameMode;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;

import io.github.flegacy.flms.FLMS;

public class WorldProtectionListener implements Listener {
	private static WorldProtectionListener instance;

	private final FLMS plugin;

	private WorldProtectionListener(FLMS plugin) {
		this.plugin = plugin;
	}

	@EventHandler
	public void onBlockBreak(BlockBreakEvent event) {
		// TODO enable compatibility with other listeners later instead of just stopping the event altogether.
		// TODO integrate crops
        if (event.getPlayer().getGameMode() != GameMode.SURVIVAL)
            return;
        if (!(event instanceof FLMSBlockBreakEvent flmsEvent)) {
            event.setCancelled(true);
            return;
        }
	}

	public static WorldProtectionListener getInstance(FLMS plugin) {
		if (instance == null)
			instance = new WorldProtectionListener(plugin);
		return instance;
	}
}

File: bin/main/io/github/flegacy/flms/mining/WorldProtectionListener.kt
package io.github.flegacy.flms.mining

import io.github.flegacy.flms.FLMS
import org.bukkit.GameMode
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent

class WorldProtectionListener(private val plugin: FLMS): Listener {

    @EventHandler
    fun onBlockBreak(event: BlockBreakEvent) {
        // TODO enable compatiiblity with other listeners later isntead of just stopping the event entirely
        // TODO integrate crops
        if (event.player.gameMode != GameMode.SURVIVAL)
            return
        val registeredCrop = plugin.registry().findBlock(event.block.type) != null && event.block.type.hardness == 0f
        if (registeredCrop)
            return
        if (event !is FLMSBlockBreakEvent) {
            event.isCancelled = true
            return
        }

    }

}
